pkg/sdk/cortex_search_services_def.go

```diff
@@ -0,0 +1,119 @@
+	WithValidation(g.AtLeastOneValueSet, "TargetLag", "Warehouse", "Comment")
+
+var CortexSearchServiceDef = g.NewInterface(
+	"CortexSearchServices",
+	"CortexSearchService",
+	g.KindOfT[SchemaObjectIdentifier](),
+).CreateOperation(
+	"https://docs.snowflake.com/LIMITEDACCESS/cortex-search/sql/create-cortex-search",
+	g.NewQueryStruct("CreateCortexSearchService").
+		// Fields
+		Create().
+		OrReplace().
+		SQL("CORTEX SEARCH SERVICE").
+		IfNotExists().
+		Name().
+		TextAssignment("ON", g.ParameterOptions().NoEquals().NoQuotes().Required().SQL("ON")).
+		OptionalQueryStructField(
+			"Attributes",
+			g.NewQueryStruct("Attributes").
+				SQL("ATTRIBUTES").
+				List("Columns", "string", g.ListOptions().NoEquals().NoParentheses()),
+			g.KeywordOptions(),
+		).
+		Identifier("Warehouse", g.KindOfT[AccountObjectIdentifier](), g.IdentifierOptions().Equals().Required().SQL("WAREHOUSE")).
+		TextAssignment("TARGET_LAG", g.ParameterOptions().SingleQuotes().Required()).
+		OptionalComment().
+		PredefinedQueryStructField("QueryDefinition", "string", g.ParameterOptions().NoEquals().NoQuotes().Required().SQL("AS")).
+		// Validations
+		WithValidation(g.ValidIdentifier, "name").
+		WithValidation(g.ValidIdentifier, "name").
+		WithValidation(g.ValidateValueSet, "On").
+		WithValidation(g.ValidateValueSet, "TargetLag").
+		WithValidation(g.ConflictingFields, "OrReplace", "IfNotExists"),
+).AlterOperation(
+	"https://docs.snowflake.com/LIMITEDACCESS/cortex-search/sql/alter-cortex-search",
+	g.NewQueryStruct("AlterCortexSearchService").
+		// Fields
+		Alter().
+		SQL("CORTEX SEARCH SERVICE").
+		IfExists().
+		Name().
+		OptionalQueryStructField("Set", alterServiceSet, g.KeywordOptions().SQL("SET")).
+		// Validations
+		WithValidation(g.ValidIdentifier, "name").
+		WithValidation(g.ExactlyOneValueSet, "Set"),
+).ShowOperation(
+	"https://docs.snowflake.com/LIMITEDACCESS/cortex-search/sql/show-cortex-search",
+	// Fields
+	g.DbStruct("cortexSearchServiceRow").
+		Field("created_on", "time.Time").
+		Field("name", "string").
+		Field("database_name", "string").
+		Field("schema_name", "string").
+		Field("comment", "string"),
+	g.PlainStruct("CortexSearchService").
+		Field("CreatedOn", "time.Time").
+		Field("Name", "string").
+		Field("DatabaseName", "string").
+		Field("SchemaName", "string").
+		Field("Comment", "string"),
+	g.NewQueryStruct("ShowCortexSearchService").
+		Show().
+		SQL("CORTEX SEARCH SERVICES").
+		OptionalLike().
+		OptionalIn().
+		OptionalStartsWith().
+		OptionalLimitFrom(),
+).ShowByIdOperation().DescribeOperation(
+	g.DescriptionMappingKindSingleValue,
+	"https://docs.snowflake.com/LIMITEDACCESS/cortex-search/sql/desc-cortex-search",
+	g.DbStruct("cortexSearchServiceDetailsRow").
+		Field("name", "string").
+		Field("schema", "string").
+		Field("database", "string").
+		Field("warehouse", "string").
+		Field("target_lag", "string").
+		Field("search_column", "string").
+		Field("included_columns", "*string").
```

Comment:
How should I setup an optional array of strings here?

Reply:
This is for an optional string array field - is there something better than `OptionalText`?

Reply:
I believe in this case for g.DbStruct has to be OptionalText (because this column is a text column, but contains a comma-separated list of column names), but in PlainStruct it should be a slice. The conversion should happen in the convert method where you have to split on a comma to get the list of columns.

Reply:
Cool - done